extensions/sql-migration/src/dialog/assessmentResults/savedAssessmentDialog.ts

```diff
@@ -0,0 +1,140 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as constants from '../../constants/strings';
+import { MigrationStateModel } from '../../models/stateMachine';
+import { WizardController } from '../../wizard/wizardController';
+
+
+export class SavedAssessmentDialog {
+
+	private static readonly OkButtonText: string = 'Next';
+	private static readonly CancelButtonText: string = 'Cancel';
+
+	private _isOpen: boolean = false;
+	private dialog: azdata.window.Dialog | undefined;
+	private _rootContainer!: azdata.FlexContainer;
+	private stateModel: MigrationStateModel;
+	private context: vscode.ExtensionContext;
+
+	constructor(context: vscode.ExtensionContext, stateModel: MigrationStateModel) {
+		this.stateModel = stateModel;
+		this.context = context;
+	}
+
+	private async initializeDialog(dialog: azdata.window.Dialog): Promise<void> {
+		return new Promise<void>((resolve, reject) => {
+			dialog.registerContent(async (view) => {
+				try {
+					this._rootContainer = this.initializePageContent(view);
+					view.initializeModel(this._rootContainer);
+					resolve();
+				} catch (ex) {
+					reject(ex);
+				}
+			});
+		});
+	}
+
+	public async openDialog(dialogName?: string) {
+		if (!this._isOpen) {
+			this._isOpen = true;
+			this.dialog = azdata.window.createModelViewDialog('Saved Assessment Result', 'Saved Assessment Result', '60%');
+
+			this.dialog.okButton.label = SavedAssessmentDialog.OkButtonText;
+			this.dialog.okButton.onClick(async () => await this.execute());
```

Comment:
please add all "on{Event}" declaration disposable to a similar array and disposable pattern like this:
				this._disposables.push(this._view.onClosed(e => {
					this._disposables.forEach(
						d => { try { d.dispose(); } catch { } });
				}));

Reply:
similar to migrationCutoverDialog.ts